core/server/common/src/main/java/alluxio/master/journal/raft/JournalStateMachine.java

```diff
@@ -98,6 +99,7 @@
   private volatile long mNextSequenceNumberToRead = 0;
   private volatile boolean mSnapshotting = false;
   private volatile boolean mIsLeader = false;
+  private volatile Runnable mInterruptCallback;
```

Comment:
I am confused as to what this is supposed to be. It is called "interrupt callback", but it is set in `suspend` and used in `pause`. I don't understand the flow of interaction. Could you explain what this is?

Reply:
The callback is set in `suspend()` by backup worker before taking a backup, and invoked in `pause()` by the state machine before reloading the states from a snapshot.

The interrupt callback is used to inform the caller of the `suspend()` method that the suspension is interrupted and the metadata states are about to be changed. This is needed because the follower cannot wait for the long running backup process to complete and resume the state machine. It has to reload immediately to avoid further downtime. 

The overall flow is:

1. Backup worker suspends state machine before backup, passing in the callback.
2. Backup worker writes journal entries to UFS
3. Raft state machine downloads a new snapshot from leader.
4. Raft state machine transitions to PAUSE state and invokes the callback.
5. Backup worker handles the callback and interrupts the backup tasks.
6. Raft state machine starts reloading the states.
7. Raft state machine finished the reload and transitions back to RUNNING state.

Reply:
I see. Can you add some of this in the comments?

Also, is there a scenario where the new snapshots always interrupt backups, thus never taking a backup?

Reply:
I added a few comments to explain the workflow.

The interruption only happens after a follower installs a snapshot from master due to falling behind the earliest journal entry available. This only occur when a new master joins, or a master has not been in contact with the quorum due to downtime or outage. It will not occur under common situation.

Reply:
ok